mmv1/third_party/terraform/services/storage/resource_storage_bucket_object_test.go

```diff
@@ -415,6 +416,50 @@ func TestAccStorageObject_holds(t *testing.T) {
 	})
 }
 
+func TestAccStorageObject_retention(t *testing.T) {
+	t.Parallel()
+
+	bucketName := acctest.TestBucketName(t)
+	data := []byte(content)
+	h := md5.New()
+	if _, err := h.Write(data); err != nil {
+		t.Errorf("error calculating md5: %v", err)
+	}
+	dataMd5 := base64.StdEncoding.EncodeToString(h.Sum(nil))
+	testFile := getNewTmpTestFile(t, "tf-test")
+	if err := ioutil.WriteFile(testFile.Name(), data, 0644); err != nil {
+		t.Errorf("error writing file: %v", err)
+	}
+	expire2Days := time.Now().UTC().Add(time.Hour * 48).Round(time.Millisecond).Format(time.RFC3339Nano)
+	expire3Days := time.Now().UTC().Add(time.Hour * 72).Round(time.Millisecond).Format(time.RFC3339Nano)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccStorageObjectDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testGoogleStorageBucketsObjectRetention(bucketName, expire2Days),
```

Comment:
Sorry to make you roll this back, but it turns out that this causes non-determinism, which is a bigger problem than us needing to update this test in 17 years. 🤦  maybe there's a way to handle this but I don't think you need to block on solving it - could you go back to the `2040-01-01` version?